niceml/data/datasets/dfdataset.py

```diff
@@ -121,24 +256,68 @@ def __len__(self):
     def on_epoch_end(self):
         """shuffles on epoch end"""
         if self.shuffle:
-            random.shuffle(self.index_list)
+            self.index_list = self.data_shuffler.shuffle(
+                data_infos=self.get_all_data_info()
+            )
 
     def iter_with_info(self):
+        """
+        The iter_with_info function is a generator that yields the next batch of data,
+        along with some additional information about the batch.
+        The additional information is useful for various diagnostic purposes.
+        The function returns an object of type DataIterator, which has two fields:
+            * 'batch' contains the next batch of data.
+            * 'info' contains additional information about that batch.
+
+        Returns:
+            A dataiterator object
+        """
         return DataIterator(self)
 
     def get_datainfo(self, batch_index):
         """creates datainfo for given index"""
```

Comment:
This function returns a list of data infos from a given batch. Please adjust the docstring to represent this.

Suggestion:
Returns a datainfo list of all data in the given batch

Reply:
I've rewritten the docstring